ext/drmgr/drmgr.c

```diff
@@ -1668,6 +1705,75 @@ drmgr_unregister_opcode_instrumentation_event(drmgr_opcode_insertion_cb_t func,
     return drmgr_bb_cb_remove(opcode_cb_list, (void *)func, cb_entry_matches_opcode);
 }
 
+DR_EXPORT
+bool
+drmgr_register_bb_post_instru_event(drmgr_xform_cb_t func, drmgr_priority_t *priority)
+{
+    if (func == NULL)
+        return false; /* invalid params */
+    return drmgr_bb_cb_add(&cblist_post_instru, (void *)func, NULL, priority,
+                           NULL /* no user data */, cb_entry_set_fields_xform);
+}
+
+DR_EXPORT
+bool
+drmgr_unregister_bb_post_instru_event(drmgr_xform_cb_t func)
+{
+    if (func == NULL)
+        return false; /* invalid params */
+    return drmgr_bb_cb_remove(&cblist_post_instru, (void *)func, cb_entry_matches_xform);
+}
+
+DR_EXPORT
+bool
+drmgr_register_bb_instrumentation_all_events(drmgr_app2app_ex_cb_t app2app_func,
+                                             drmgr_ilist_ex_cb_t analysis_func,
+                                             drmgr_insertion_cb_t insertion_func,
+                                             drmgr_ilist_ex_cb_t instru2instru_func,
+                                             drmgr_ilist_ex_cb_t post_instru_func,
+                                             drmgr_priority_t *priority)
+{
+    if (!drmgr_register_bb_instrumentation_ex_event(
+            app2app_func, analysis_func, insertion_func, instru2instru_func, priority))
+        return false;
+
+    if (post_instru_func == NULL)
+        return false;
+
+    bool ok = true;
+    if (post_instru_func != NULL) {
+        ok = drmgr_bb_cb_add(&cblist_post_instru, (void *)post_instru_func, NULL,
+                             priority, NULL /* no user data */,
```

Comment:
You mentioned that we need `drmgr_register_bb_instrumentation_all_events` as we want to pass user data between all five stages. But here it says "no user data". Am I understanding something wrong?